storage/innobase/fts/fts0fts.cc

```diff
@@ -3325,8 +3325,11 @@ fts_fetch_doc_from_rec(
 					static_cast<mem_heap_t*>(
 						doc->self_heap->arg));
 		} else {
-			doc->text.f_str = (byte*) rec_get_nth_field(
-				clust_rec, offsets, clust_pos,
+			ut_ad(rec_offs_validate(clust_rec, clust_index, offsets));
+
+			doc->text.f_str = (byte*) rec_get_nth_cfield(
+				clust_rec, offsets, clust_pos, clust_index,
+				static_cast<mem_heap_t*>(doc->self_heap->arg),
 				&doc->text.f_len);
```

Comment:
Instant ADD COLUMN is never invoked on the internally created tables that implement FULLTEXT INDEX for InnoDB tables. This change can and should be reverted.